core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/QuicksilverMigrationMonitorSupport.scala

```diff
@@ -0,0 +1,129 @@
+          currentValue match {
+            // current attribute does not exist
+            case None               => None
+            case Some(currentValue) => Some((attributeName, compareAttrs(currentValue, correctionValue)))
+          }
+      }
+      // determine overall entity status from attribute statuses
+      val entityStatus = calculateEntityStatus(attrComparisons)
+      (entityStatus, attrComparisons)
+  }
+
+  def compareAttrs(current: Attribute, correction: Attribute): AttributeCorrectionStatusType =
+    (current, correction) match {
+      // current attribute is not the same type as correction attribute
+      case _ if current.getClass != correction.getClass =>
+        AttributeCorrectionStatus.TypeDifferent
+      case (x: AttributeValueList, y: AttributeValueList)                     => compareLists(x, y)
+      case (x: AttributeEntityReferenceList, y: AttributeEntityReferenceList) => compareLists(x, y)
+      case _                                                                  =>
+        throw new Exception(
+          s"unexpected classes found! current: ${current.getClass.getName}; correction: ${correction.getClass.getName}"
+        )
+    }
+
+  def compareLists(current: AttributeValueList, correction: AttributeValueList): AttributeCorrectionStatusType =
+    compareLists[AttributeValue](current.list, correction.list)
+
+  def compareLists(current: AttributeEntityReferenceList,
+                   correction: AttributeEntityReferenceList
+  ): AttributeCorrectionStatusType =
+    compareLists[AttributeEntityReference](current.list, correction.list)
+
+  // inspect the elements of the lists
+  def compareLists[T](current: Seq[T], correction: Seq[T]): AttributeCorrectionStatusType = {
+    val sameSize = current.size == correction.size
+
+    if (sameSize) {
+      if (current == correction) {
+        // exactly equal; this is correct
+        AttributeCorrectionStatus.Correct
+      } else {
+        // generate a map of AttributeValue -> count, where $count is the number of times that element appears
+        // in the list
+        val currentCounts = current.groupBy(identity).map { case (attrVal, seq) => (attrVal, seq.size) }
+        val correctionCounts = correction.groupBy(identity).map { case (attrVal, seq) => (attrVal, seq.size) }
+        // compare the elements and their counts; this is an unordered comparison
+        if (currentCounts == correctionCounts) {
+          // list has the same elements, and in the same counts, but did not pass the
+          // current.list == correction.list comparison; it must have been reordered
+          AttributeCorrectionStatus.Reordered
+        } else {
+          // list has different elements, or some of those elements have different counts
+          AttributeCorrectionStatus.Different
+        }
+      }
+    } else {
+      if (current.startsWith(correction)) {
+        AttributeCorrectionStatus.StartsWith
```

Comment:
Is this trying to detect the situation where the list wasn't reordered, but users have since added new values to the list since the quicksilver migration ran? If so, should we be checking in the other direction to find lists that weren't reordered by quicksilver but have been truncated? Or is that just not a common pattern worth handling here?